app/instrumentation-tests/src/org/commcare/AsyncRestoreHelperMock.java

```diff
@@ -0,0 +1,44 @@
+package org.commcare;
+
+import org.commcare.network.RemoteDataPullResponse;
+import org.commcare.tasks.AsyncRestoreHelper;
+import org.commcare.tasks.DataPullTask;
+import org.commcare.tasks.ResultAndError;
+
+/**
+ * @author $|-|!˅@M
+ */
+public class AsyncRestoreHelperMock extends AsyncRestoreHelper {
```

Comment:
I think the reason why I couldn't use mockito is because I have no way to inject the mock [here](https://github.com/dimagi/commcare-android/pull/2231/files#diff-5ce2bf85747c398fb9c88402a8a693b2L109)

Reply:
is there a way we can mock the `DataPullTask` itself instead of `AsyncRestoreHelper` to check for this behaviour ? `AsyncRestoreHelper` is in an internal api to `DataPullTask` and should ideally be treated as a black box.

Reply:
Ohh, so it'll be like: I mock `DataPulLtask` and then override its constructor to use `AsyncRestoreHelperMock`. And then perhaps maybe I'll have to mock `FormAndDataSyncer` as well and override it's method to use `DataPullTaskMock`? Or maybe I'm wrong here?

Reply:
Yeah you can make 2 changes - 
1.  Change `FormAndDataSyncer` to utilize a `DataPullTaskFactory` and then override that factory class in the tests to return you the mocked class. 
2. Have a setter for `asyncRestoreHelper` in `DataPullTask` to set the mocked `asyncRestoreHelper`

Reply:
Also realized you can also just introduce a factory for `asyncRestoreHelper` and override that in tests as well, Basically I am trying to avoid having any test specific functions in non-test code.